docs/workflow/wasm-documentation.md

```diff
@@ -0,0 +1,232 @@
+- **[WASM Runtime Debugging](debugging/mono/wasm-debugging.md)** - Debug the Mono runtime, native crashes, and collect stack traces
+- **[VS Code Debugging](debugging/libraries/debugging-vscode.md)** - Set up VS Code for debugging WASM applications
+
+### Common Debugging Scenarios
+
+#### Debug with VS Code
+Add this configuration to your `.vscode/launch.json`:
+```json
+{
+    "name": "Libraries",
+    "request": "attach",
+    "type": "chrome",
+    "address": "localhost",
+    "port": <PROXY_PORT>
+}
+```
+
+See detailed instructions in:
+- [Library debugging guide](testing/libraries/debugging-wasm.md#debug-with-vs-code)
+- [WASI debugging](../src/mono/wasi/README.md#4-debug-it)
+- [VS Code debugging guide](debugging/libraries/debugging-vscode.md)
+
+#### Debug with Chrome DevTools
+1. Open `chrome://inspect/#devices` in Chrome
+2. Configure the proxy address from your test output
+3. Select "Inspect" on your application
+4. Set breakpoints in the Sources tab under `file://`
+
+## Features and Configuration
+
+### Runtime Features
+- **[WebAssembly Features](../../src/mono/wasm/features.md)** - Configure browser features, SIMD, threads, AOT, and more
+- **[Threading Support](../../src/mono/wasm/threads.md)** - Multi-threading support and limitations
+
+### Build Configuration
+Key MSBuild properties for WebAssembly applications:
+- `WasmEnableThreads` - Enable multi-threading support
+- `WasmEnableSIMD` - Enable SIMD instruction support  
+- `RunAOTCompilation` - Enable Ahead-of-Time compilation
+- `WasmBuildNative` - Force native rebuild
+- `EnableDiagnostics` - Enable diagnostic features
+
+### Globalization and ICU
+- **[ICU for WebAssembly](../../design/features/globalization-icu-wasm.md)** - Globalization and ICU database configuration
+
+## Deployment and Hosting
+
+### Application Structure
+- **AppBundle folder**: Contains your deployed application
+- **_framework folder**: Runtime and managed assemblies
+- **dotnet.js**: Main entry point and JavaScript API
+- **dotnet.native.wasm**: Compiled .NET runtime binary
+
+### Web Server Configuration
+Ensure your web server is configured with proper MIME types:
+- `.wasm` files: `application/wasm`
+- `.js` files: `text/javascript`
+- `.json` files: `application/json`
+
+## Advanced Topics
+
+### Performance and Optimization
+- **Profiling**: Use browser dev tools profiler integration
+- **AOT Compilation**: Improve runtime performance with ahead-of-time compilation
+- **IL Trimming**: Reduce application size by removing unused code
+
+### Samples and Examples
+Located in `src/mono/sample/wasm/`:
+- **browser-bench**: Performance benchmarking sample
+- **browser-profile**: Profiling sample
+- **console**: Console application samples
+
+### Workloads
+- `wasm-tools`: Production WebAssembly tools and optimization
+- `wasm-experimental`: Experimental features and templates
+
+## FAQ
+
+### How do I debug a library test failure seen on CI?
+
+1. **Run the test locally with debugging enabled**:
```

Comment:
Do not add instructions in FAQ if the answer is already in another doc. Just link the doc section as the answer.

Review all FAQ for that.